### ISSUE TITLE
Asking for the pretrained Netvlad model

### ISSUE THREAD

**Issue by Jemmagu** [mpkuse/cartwheel_train#6]
Hi @mpkuse , 
thank you for you contribution. And I'm writing to ask for you pretrained Netvlad model because it's said available on the page.
Looking forward to your reply, thanks in advance!

Best,
Jemma

**Comment by mpkuse** [comment]
Hi,
I am sorry the pretrained models are not well organized. Some of the models which I am currently using are available here: https://github.com/mpkuse/cerebro/tree/master/scripts/keras.models 

The models in June2019 are what I use for placerecognition in the SLAM system. 

Hope this helps

**Comment by Jemmagu** [comment]
Thanks, but I only need the Netvlad part, can I extract it from the whole model?

**Comment by Jemmagu** [comment]
Hi, the output of the models in June2019 seems to be a 1024-d global descriptor, so can I use it directly for image retrieval?

Thanks in advance!

**Comment by mpkuse** [comment]
If you really need NetVLAD, try the original NETVLAD code: https://github.com/Relja/netvlad . If you wish to convert those weights for use with keras have a look at this script I had written up: https://github.com/mpkuse/cartwheel_train/blob/master/test_relja.py 

Or else, you can train everything my yourself using my code, by setting the back end net as VGG16 and K=64, which is the most similar to NetVLAD.

**Comment by mpkuse** [comment]
It is also possible to use thr 1024-D descriptor for image retrival, but performance might not be as good as the one with VGG. I am using the 1024-dim descriptor for SLAM system loop detection. It works for such cases.

**Comment by Jemmagu** [comment]
Actullay, I've tried the original Netvlad to do image retrieval. But it seems your modified Netvlad achieves similar results with lower time cost. That's why I wanna try your Netvlad. So can your Netvlad achieve similar results with lower time when doing image retrieval?

**Comment by mpkuse** [comment]
Not tried extensively with a larger general image retrieval. I have only tested with the SLAM like datasets. But I think if you train mobilenet (like I do) instead of VGG and tune some params and thoroughly do testing it can be made to work for image retrieval task as well which will be much quicker than original NetVLAD.
